File: generate-cda/generate-cda.go
package main

import (
	"fmt"
	"github.com/iotaledger/iota.go/account/builder"
	"github.com/iotaledger/iota.go/account/deposit"
	"github.com/iotaledger/iota.go/account/store/badger"
	"github.com/iotaledger/iota.go/account/timesrc"
	"github.com/iotaledger/iota.go/api"
	"github.com/iotaledger/iota.go/trinary"
	"time"
)

// You should never hard-code a seed
var seed = trinary.Trytes("PUETTSEITFEVEWCWBTSIZM9NKRGJEIMXTULBACGFRQK9IMGICLBKW9TTEVSDQMGWKBXPVCBMMCXWMNPDX")

func main() {
	// Define the node to connect to
	apiSettings := api.HTTPClientSettings{URI: "https://nodes.devnet.iota.org:443"}

	iotaAPI, err := api.ComposeAPI(apiSettings)
	handleErr(err)

	store, err := badger.NewBadgerStore("db")
	handleErr(err)

	// Make sure the database closes when the code stops
	defer store.Close()

	timesource := timesrc.NewNTPTimeSource("time.google.com")

	account, err := builder.NewBuilder().
		// Connect to a node
		WithAPI(iotaAPI).
		// Create the database
		WithStore(store).
		WithSeed(seed).
		// Set the minimum weight magnitude for the Devnet
		WithMWM(9).
		WithTimeSource(timesource).
		// Load the default plugins that enhance the functionality of the account
		WithDefaultPlugins().
		Build()
	handleErr(err)

	handleErr(account.Start())

	// Make sure the account shuts down when the code stops
	defer account.Shutdown()

	// Get the current time
	now, err := timesource.Time()
	handleErr(err)

	// Define the same time tomorrow
	now = now.Add(time.Duration(24) * time.Hour)

	// Specify the conditions
	conditions := &deposit.Conditions{TimeoutAt: &now, MultiUse: true}

	// Generate the CDA
	cda, err := account.AllocateDepositAddress(conditions)
	handleErr(err)

	fmt.Println(cda.AsMagnetLink())

}

func handleErr(err error) {
	if err != nil {
		panic(err)
	}
}

File: java/account-module/src/main/java/com/iota/GenerateCDA.java
package com.iota;

import java.io.File;
import java.util.Date;
import java.util.concurrent.ExecutionException;

import org.iota.jota.IotaAPI;
import org.iota.jota.IotaAccount;
import org.iota.jota.account.AccountStore;
import org.iota.jota.account.deposits.ConditionalDepositAddress;
import org.iota.jota.account.deposits.methods.DepositFactory;
import org.iota.jota.account.deposits.methods.MagnetMethod;
import org.iota.jota.account.errors.AccountError;
import org.iota.jota.account.store.AccountFileStore;
import org.iota.jota.error.ArgumentException;

class GenerateCDA {
    public static void main(String[] args)
            throws ArgumentException, AccountError, ExecutionException, InterruptedException {

        // Connect to a node
        IotaAPI api = new IotaAPI.Builder()
            .protocol("https")
            .host("nodes.devnet.thetangle.org")
            .port(443)
            .build();
    
        // The seed that the account uses to generate CDAs and send bundles
        String mySeed = "PUETTSEITFEVEWCTBTSIZM9NKRGJEIMXTULBACGFRQK9IMGICLBKW9TTEVSDQMGWKBXPVCBMMCXWMNPDX";
        
        // Create a file to store the seed state
        File file = new File("seed-state-database.json");
        AccountStore store = new AccountFileStore(file);
        
        // Create an account, using your seed
        IotaAccount account = new IotaAccount.Builder(mySeed)
            // Connect to a node
            .api(api)
            // Connect to the database
            .store(store)
            // Set the minimum weight magnitude for the Devnet (default is 14)
            .mwm(9)
            // Set a security level for CDAs (default is 3)
            .securityLevel(2)
            .build();
    
        // Start the account and any plugins
        account.start();
    
        // Define the same time tomorrow
        Date timeoutAt = new Date(System.currentTimeMillis() + 24000 * 60 * 60);
    
    	// Generate the CDA
    	ConditionalDepositAddress cda = account.newDepositAddress(timeoutAt, true, 0).get();
    
        String magnet = (String) DepositFactory.get().build(cda, MagnetMethod.class);
        
        System.out.println(magnet);
        
        // Close the database and stop any ongoing reattachments
        account.shutdown();
    }
}

